keyboards/tetris/keymaps/default/keymap.c

```diff
@@ -1,829 +1,285 @@
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS),
+
+  KEYMAP(
+    RGB_M_T, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, RESET,
+    KC_TRNS, RGB_M_P, RGB_M_B, RGB_M_R, RGB_M_SW, RGB_M_SN, RGB_M_K, RGB_M_X, RGB_M_G, KC_TRNS, KC_TRNS, KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, RGB_VAI, KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, RGB_TOG, KC_TRNS, KC_TRNS, RGB_RMOD, RGB_VAD, RGB_MOD,
+    RGB_VAD, RGB_VAI, RGB_VAD, RGB_VAI)
 };
 
-/* Encoder */
-static uint8_t  encoder_state  = 0;
-static int8_t  encoder_value  = 0;
-static int8_t  encoder_LUT[]  = { 0, -1, 1, 0, 1, 0, 0, -1, -1, 0, 0, 1, 0, 1, -1, 0 };
-
-/* Timer */
-static uint16_t TAP_RGB_timer;
+extern rgblight_config_t rgblight_config;
+int RGB_LAYER0_mode = 6;
+static uint8_t current_layer = 0;
+static bool has_layer_changed = true;
+uint16_t i = 0;
+uint16_t j = 0;
 static uint16_t RGB_encoder_timer;
+static uint16_t RGB_encoder_timer2;
+static uint8_t RGB_encoder_dir = 0;
 
-/* Encoder state RGB_display */
-static uint8_t  RGB_encoder_count=6;
-static bool  RGB_encoder_bool  = false;
-
-/* RGB tap HUE */
-static uint16_t  Type_Hue=270;
-
-void matrix_init_user( void )
-{
-  wait_ms( 200 );
-
-  /* Encoder init */
-  encoder_state = PIND & 0x3;
+const uint8_t RGBLED_RAINBOW_MOOD_INTERVALS[] PROGMEM = {
+  30,
+  5,
+  -5
+};
 
-  /* Timer init */
-  RGB_encoder_timer= timer_read();
-  TAP_RGB_timer= timer_read();
+void matrix_init_user(void) {
+  RGB_encoder_timer = timer_read();
+  RGB_encoder_timer2 = timer_read();
 }
 
+void matrix_scan_user(void) {
+  uint8_t layer = biton32(layer_state);
 
-void matrix_scan_user( void )
-{
-/* Layer  */
-  uint8_t layer = biton32( layer_state );
-
-/* Encoder  */
-  encoder_state  <<= 2;
-  encoder_state  |= (PIND & 0x3);
-  encoder_value  += encoder_LUT[encoder_state & 0xF];
-
-/* Encoder state RGB display */
-  if (RGB_encoder_bool){
-    if (timer_elapsed(RGB_encoder_timer) > 2500) {
-      RGB_encoder_bool=false;
-      if (RGB_TAP_STATE) {
-        rgblight_mode( 1 );
-        rgblight_setrgb( RGB_TAP_Base_Color );
-      }else{
-        rgblight_mode( RGB_LAYER0_mode );
-      }
-    }
-  }
-
-/* Display the Caps state */
-  if (caps==true){
-    rgblight_setrgb_at(RGB_Caps_Color,12);
-  }
-
-/* Start of Encoder clockwise */
-  if ( encoder_value >= 4 ) {
-
-  /* Start of Encoder state RGB_display */
-    if (!RGB_encoder_bool){
-      RGB_encoder_timer=timer_read();
-      RGB_encoder_bool=true;
-      RGB_encoder_count=5;
-      rgblight_mode(1);
-      rgblight_setrgb( 0,0,0 );
-    }else if (RGB_encoder_bool && timer_elapsed(RGB_encoder_timer) < 800) {
-      RGB_encoder_timer=timer_read();
-      RGB_encoder_count-=1;
-      if (RGB_encoder_count>5 || RGB_encoder_count<1){
-        RGB_encoder_count=5;
+  if (RGB_encoder_dir != 0) {
```

Comment:
If this is set to the default color, then you don't need this anymore. 

You can use `keyboard_post_init_user` instead, as this runs after the rgblight feature has been initialized.

Reply:
Some rgblight  animation will show up with the encoder,this codes is the animation code.